crates/typst-cli/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+pub mod args;
+pub mod compile;
+pub mod download;
+pub mod fonts;
+pub mod init;
+pub mod package;
+pub mod query;
+pub mod terminal;
+pub mod timings;
+#[cfg(feature = "self-update")]
+pub mod update;
+pub mod watch;
+pub mod world;
```

Comment:
Exposing all of these comes with a big semver hazard, if any of these APIs which are specific to typst-cli alone like `update` are changed, a new major version of the library would be required.

In theory, no one in their right mind would use the `update` module, but this is not enough of an argument to ignore the semver hazard I'm afraid.